infra/vsts_agent_linux_startup.sh

```diff
@@ -95,6 +96,41 @@ useradd \
 #add docker group to user
 usermod -aG docker vsts
 
+# The Bazel cache fills up the hard drive in less than a day, and is a real
+# pain to clean up, as it consists of a very large amount of read-only files in
+# a fairly large tree of read-only directories. Therefore, it looks like the
+# best way to manage it is to mount it on a partition and reset the partition.
+
+CLEANUP=/home/vsts/cache_cleanup.sh
+cat <<'CACHE_CLEANUP' > $CLEANUP
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+local_cache=$HOME/.cache/bazel
+disk_cache=$HOME/.bazel-cache
+
+for file in local_cache disk_cache; do
+    eval path='$'$file
+    echo "$(date -Is) Cleaning up '$path'..."
+    if [ -d "$path/lost+found" ]; then
+        fusermount -u "$path"
+    fi
+    rm -f "$HOME/$file"
+    truncate -s 200g "$HOME/$file"
+    mkfs.ext2 -E root_owner=$UID:$GID "$HOME/$file"
```

Comment:
I decided on ext2 over ext3 because we don't need journaling. I must admit I'm not clear on what the specific benefits of ext4 are; from what I've (very quickly) skimmed it seems to be mostly adding support for very large files and very large filesystems, neither of which we need here.

Reply:
sounds reasonable, I don’t know the details either, I just tend to default to ext4 these days. Not needing journaling sounds like a good reasons not to use that.